kiwi/builder/live.py

```diff
@@ -146,93 +147,80 @@ def create(self) -> Result:
         log.info(
             'Setting up live image bootloader configuration'
         )
-        if self.firmware.efi_mode():
-            # setup bootloader config to boot the ISO via EFI
-            # This also embedds an MBR and the respective BIOS modules
-            # for compat boot. The complete bootloader setup will be
-            # based on grub
-            bootloader_config = BootLoaderConfig.new(
-                self.bootloader, self.xml_state, root_dir=self.root_dir,
-                boot_dir=self.media_dir.name, custom_args={
-                    'grub_directory_name':
-                        Defaults.get_grub_boot_directory_name(self.root_dir),
-                    'grub_load_command':
-                        'configfile'
-                }
+        with self._bootloader_instance() as bootloader_config:
+            if self.firmware.efi_mode():
+                bootloader_config.setup_live_boot_images(
+                    mbrid=self.mbrid, lookup_path=self.root_dir
+                )
+
+            IsoToolsBase.setup_media_loader_directory(
+                self.boot_image.boot_root_directory, self.media_dir.name,
+                bootloader_config.get_boot_theme()
             )
-            bootloader_config.setup_live_boot_images(
-                mbrid=self.mbrid, lookup_path=self.root_dir
+            if self.firmware.bios_mode():
+                Iso(self.media_dir.name).setup_isolinux_boot_path()
```

Comment:
These lines appear exactly the same way in `kiwi/builder/install.py` lines 210 & following

Reply:
yes correct and intentional because this action needs to be done for building live ISO images (builder/live) and for building install ISO image (builder/install). Those are two different builders but they both produce some sort of ISO, which is the reason why you find the same function call in both builders, is this a problem ?

btw, in the other open PR #2439 these will also disappear

Reply:
Code duplication is generally never a good idea, as changing one instance requires changing all instances.

However, if this code will go, then I think it's fine to keep it as is for now.